pkg/aspect/config/metrics.proto

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package pkg.aspect.config;
+
+option go_package="config";
+
+// Configures a metric aspect.
+message MetricsParams {
+    // Describes how attributes must be evaluated to produce values for the named metric. Suppose the following
+    // MetricDescriptor exists in the global configuration:
+    //
+    //    metric_descriptor:
+    //      name: "response_code"
+    //      kind: COUNTER
+    //      value: I64
+    //      labels:
+    //        name: api_method
+    //        value_type: STRING
+    //      labels:
+    //        name: response_code
+    //        value_type: I64
+    //
+    // To actually report metrics at run time a mapping from attributes to a metric's labels must be provided in
+    // the form of a metric:
+    //
+    //    metric:
+    //      descriptor: "response_code" # must match metric_descriptor.name
+    //      value: $requestCount        # Istio expression syntax for the attribute named "requestCount"
+    //      labels:
+    //        # either the attribute named 'apiMethod' or the literal string 'unknown'; must eval to a string
+    //        api_method: $apiMethod | "unknown"
+    //        # either the attribute named 'responseCode' or the literal int64 500; must eval to an int64
+    //        response_code: $responseCode | 500
+    message Metric {
+        // Must match the name of some metric_descriptor in the global config.
+        string descriptor = 1;
+
+        // Attribute expression to evaluate to determine the value for this metric;
+        // the result of the evaluation must match the value ValueType of the metric_descriptor.
+        string value = 2;
```

Comment:
can we call this `valueExpression` and  `labelExpression` ?
names are useful when people don't  RTFM. 

Another thing to note here is that the expression evaluator is only aware of `ValueTypes`. So it can correctly process and produces the supported value types.

I don't think labelTypes indirection is giving us a whole lot. We will need to map it down multiple times. 
I think it is ok to has   
`func isValidlableType(ValueType) {}` to subset it.